caddyhttp/httpserver/middleware.go

```diff
@@ -48,20 +47,48 @@ type (
 
 	// RequestMatcher checks to see if current request should be handled
 	// by underlying handler.
-	//
-	// TODO The long term plan is to get all middleware implement this
-	// interface and have validation done before requests are dispatched
-	// to each middleware.
 	RequestMatcher interface {
 		Match(r *http.Request) bool
 	}
+
+	// Config is a middleware configuration.
+	// This makes it possible for middlewares to have a common
+	// configuration interface.
+	//
+	// TODO The long term plan is to get all middleware implement this
+	// interface for configurations.
+	Config interface {
+		RequestMatcher
```

Comment:
The name `httpserver.Config` is a little ambiguous and looks more like a config for an HTTP server; maybe name this type `MiddlewareConfig` or `HandlerConfig` instead (I think HandlerConfig is technically more accurate, but I'm not married to it if you disagree)?